File: gunhoflash/valid-palindrome.ts
function isPalindrome(s: string): boolean {
  s = s.replace(/[^0-9A-Za-z]/g, '').toLowerCase();
  const length = s.length;
  const half = (length + 1) >> 1;
  for (let i = 0; i < half; i++) {
    if (s[i] !== s[length - 1 - i]) {
      return false;
    }
  }
  return true;
}

File: gunhoflash/valid-palindrome.py
import re

class Solution:
  def isPalindrome(self, s: str) -> bool:
    s = re.sub(r'[^0-9A-Za-z]', '', s).lower()
    length = len(s)
    half = (length + 1) // 2
    for i in range(half):
      if s[i] != s[length - 1 - i]:
        return False
    return True
